### ISSUE TITLE
CLI 0.4.8 fails to deploy project.yaml to Lightning v0.10.0-pre9

### ISSUE THREAD

**Issue by bensonmiller** [OpenFn/kit#495]
### User story
I'm running a local instance of Lightning v0.10.0-pre9. I can successfully perform an `openfn pull` against a project in my instance, but I cannot successfully `openfn deploy` that same project file back to Lightning. I'm testing this with a very simple workflow in which all jobs use `common` adaptor. I've pasted the output from failed `deploy` job at the bottom of this issue.

Other notes:

- It seems like command-line options aren't working for `-c | --config` and the `-p | --project-path` options. Setting these options has no effect and the `Deploying with config. . . ` CLI output shows no change to the default `configPath` or `specPath` variables. The same might be true of other CLI arguments.

I assume that I'm just experiencing a period where Lightning is undergoing a significant amount of change and the associated changes to the CLI are just lagging a bit behind the work on Lightning. Please let me know if additional information would be valuable. 

### Details
Lightning version: v0.10.0-pre9 (specific commit: 61e8c068c68b04f8753a80ce0d1592d7e3de5550)
CLI version: 0.4.8
NodeJS version: 18.17.1

### Failure output

```
# OPENFN_ENDPOINT=http://localhost:4000/ openfn deploy -l debug --no-confirm
[CLI] ❯ Deploying with config {
  "apiKey": "<snip>",
  "configPath": "./.config.json",
  "specPath": "project.yaml",
  "statePath": ".state.json",
  "endpoint": "http://localhost:4000/",
  "requireConfirmation": false,
  "dryRun": false
}
[CLI] ℹ Using OPENFN_API_KEY environment variable
[CLI] ℹ Using OPENFN_ENDPOINT environment variable
[CLI] ❯ Deploying with config {
  "apiKey": "<snip>",
  "configPath": "./.config.json",
  "specPath": "project.yaml",
  "statePath": ".state.json",
  "endpoint": "http://localhost:4000/",
  "requireConfirmation": false,
  "dryRun": false
}
[CLI] ℹ Deploying
[CLI] ❯ spec {
  "errors": [],
  "doc": {
    "name": "mytest",
    "workflows": {
      "ccdx-demo": {
        "name": "ccdx-demo",
        "jobs": {
          "attach-json-schema": {
            "name": "attach-json-schema",
            "adaptor": "@openfn/language-common@latest",
            "body": "const schema = {\n  \"$schema\": \"http://json-schema.org/draft-07/schema#\",\n  \"type\": \"object\",\n  \"required\": [\n      \"AMFR\",\n      \"AMOD\",\n      \"ASER\",\n      \"LMFR\",\n      \"LMOD\",\n      \"LSER\",\n      \"TXID\",\n      \"TSRC\",\n  \"TSCV\"\n  ],\n  \"properties\": {\n    \"ABST\": {\n      \"type\": \"string\"\n    },\n    \"AMFR\": {\n      \"type\": \"string\"\n    },\n    \"AMOD\": {\n      \"type\": \"string\"\n    },\n    \"ASER\": {\n      \"type\": \"string\"\n    },\n    \"LMFR\": {\n      \"type\": \"string\"\n    },\n    \"LMOD\": {\n      \"type\": \"string\"\n    },\n    \"LSER\": {\n      \"type\": \"string\"\n    },\n    \"TXID\": {\n      \"type\": \"string\"\n    },\n    \"TSCV\": {\n      \"type\": \"string\"\n    },\n    \"TSRC\": {\n      \"type\": \"string\"\n    }\n  }\n};\n\nfn(state =>{\n  state.schema = schema;\n  return state;\n})\n"
          },
          "validate-json-schema": {
            "name": "validate-json-schema",
            "adaptor": "@openfn/language-common@latest",
            "body": "// Get started by adding operations from the API reference\nexecute(\n  // Validate JSON schema (from state.schema) against state.data\n  validate(),\n\n  // Then send validation down success/failure path\n  fn(state => {\n    if (state.validationErrors.length > 0) {\n      state.valid = false;\n      throw new Error('Failed to validate input schema.');\n    } else {\n      state.valid = true;\n    }\n    return state;\n  })\n)\n"
          },
          "handle-valid": {
            "name": "handle-valid",
            "adaptor": "@openfn/language-common@latest",
            "body": "// Get started by adding operations from the API reference\nfn(state => {\n  return state;\n})\n"
          },
          "handle-invalid": {
            "name": "handle-invalid",
            "adaptor": "@openfn/language-common@latest",
            "body": "// Get started by adding operations from the API reference\nfn(state => {\n  return state;\n})\n"
          }
        },
        "triggers": {
          "webhook": {
            "type": "webhook"
          }
        },
        "edges": {
          "webhook->attach-json-schema": {
            "source_trigger": "webhook",
            "target_job": "attach-json-schema",
            "condition": "always",
            "enabled": true
          },
          "attach-json-schema->validate-json-schema": {
            "source_job": "attach-json-schema",
            "target_job": "validate-json-schema",
            "condition": "on_job_success",
            "enabled": true
          },
          "validate-json-schema->handle-valid": {
            "source_job": "validate-json-schema",
            "target_job": "handle-valid",
            "condition": "on_job_success",
            "enabled": true
          },
          "validate-json-schema->handle-invalid": {
            "source_job": "validate-json-schema",
            "target_job": "handle-invalid",
            "condition": "on_job_failure",
            "enabled": true
          }
        }
      }
    }
  }
}
[CLI] ℹ Getting project from server...
Checking http://localhost:4000/api/provision/1717e6d1-81f8-4225-bcfc-88a83c8f92ae for existing project.
Project found.
[CLI] ❯ currentProject 
{
  id: '1717e6d1-81f8-4225-bcfc-88a83c8f92ae',
  name: 'mytest',
  description: null,
  updated_at: '2023-11-16T18:19:31',
  scheduled_deletion: null,
  inserted_at: '2023-11-16T18:19:31',
  requires_mfa: false,
  workflows: [
    {
      id: '0c7f58aa-10c6-424a-b9a9-d8fc2f463112',
      name: 'ccdx-demo',
      deleted_at: null,
      project_id: '1717e6d1-81f8-4225-bcfc-88a83c8f92ae',
      updated_at: '2023-11-16T18:42:52',
      inserted_at: '2023-11-16T18:21:15',
      edges: [Array],
      jobs: [Array],
      triggers: [Array]
    }
  ]
}
[CLI] ❯ nextProject 
{
  id: '1717e6d1-81f8-4225-bcfc-88a83c8f92ae',
  name: 'mytest',
  description: null,
  inserted_at: '2023-11-16T18:19:31',
  updated_at: '2023-11-16T18:19:31',
  scheduled_deletion: null,
  requires_mfa: false,
  workflows: [
    {
      id: '0c7f58aa-10c6-424a-b9a9-d8fc2f463112',
      name: 'ccdx-demo',
      inserted_at: '2023-11-16T18:21:15',
      project_id: '1717e6d1-81f8-4225-bcfc-88a83c8f92ae',
      updated_at: '2023-11-16T18:42:52',
      deleted_at: null,
      triggers: [Array],
      jobs: [Array],
      edges: [Array]
    }
  ]
}
[CLI] ♦ Changes:
 {
   workflows: [
     {
       jobs: [
         {
+          enabled: true
         }
         {
+          enabled: true
-          body: "// Get started by adding operations from the API reference\nexecute(\n  // Validate JSON schema (from state.schema) against state.data\n  validate(),\n\n  // Then send validation down success/failure path\n  fn(state => {\n    if (state.validationErrors.length > 0) {\n      state.valid = false;\n      throw new Error('Failed to validate input schema.');\n    } else {\n      state.valid = true;\n    }\n    return state;\n  })\n)"
+          body: "// Get started by adding operations from the API reference\nexecute(\n  // Validate JSON schema (from state.schema) against state.data\n  validate(),\n\n  // Then send validation down success/failure path\n  fn(state => {\n    if (state.validationErrors.length > 0) {\n      state.valid = false;\n      throw new Error('Failed to validate input schema.');\n    } else {\n      state.valid = true;\n    }\n    return state;\n  })\n)\n"
         }
         {
+          enabled: true
-          body: "// Get started by adding operations from the API reference\nfn(state => {\n  return state;\n})"
+          body: "// Get started by adding operations from the API reference\nfn(state => {\n  return state;\n})\n"
         }
         {
+          enabled: true
-          body: "// Get started by adding operations from the API reference\nfn(state => {\n  return state;\n})"
+          body: "// Get started by adding operations from the API reference\nfn(state => {\n  return state;\n})\n"
         }
       ]
     }
   ]
 }

[CLI] ✘ Failed to deploy project mytest:
{
  "errors": {
    "workflows": [
      {
        "jobs": [
          {
            "base": [
              "extraneous parameters: enabled"
            ]
          },
          {
            "base": [
              "extraneous parameters: enabled"
            ]
          },
          {
            "base": [
              "extraneous parameters: enabled"
            ]
          },
          {
            "base": [
              "extraneous parameters: enabled"
            ]
          }
        ]
      }
    ]
  }
}
```

Corresponding failure output from `iex` console:

```
Done in 9892ms.
[info] GET /api/provision/1717e6d1-81f8-4225-bcfc-88a83c8f92ae
[info] Sent 200 in 33ms
[info] POST /api/provision/
[info] Sent 422 in 60ms
```

**Comment by josephjclark** [comment]
Thank you @bensonmiller! We'll take a look and get this fixed

**Comment by josephjclark** [comment]
Hi @bensonmiller! This should be fixed in `cli@0.4.9`.

I'll close the issue for now  but please let us know if the problem persists!
